VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/UploadFile/UploadFile.tsx

```diff
@@ -78,38 +79,59 @@ function UploadFile({ navigation, route }: UploadFileProps) {
 
   useEffect(() => {
     if (filesUploadedSuccess) {
-      navigateTo('FileRequest', { claimID: claim?.id || '' })
+      navigateTo('ClaimDetailsScreen', { claimID: claimID, claimType: ClaimTypeConstants.ACTIVE })
     }
-  }, [filesUploadedSuccess, claim, navigateTo])
+  }, [filesUploadedSuccess, claimID, navigateTo])
 
   const [documentType, setDocumentType] = useState('')
   const [onSaveClicked, setOnSaveClicked] = useState(false)
   const [confirmed, setConfirmed] = useState(false)
 
   useEffect(() => {
-    setRequest((prevRequest) => {
-      return {
-        ...prevRequest,
+    if (originalRequest) {
+      setRequest({
+        ...originalRequest,
         documentType,
-      }
-    })
-  }, [documentType])
+      })
+    }
+  }, [documentType, originalRequest])
 
   const onUploadConfirmed = () => {
-    logAnalyticsEvent(Events.vama_evidence_cont_3(claim?.id || '', request.trackedItemId || null, request.type, 'file'))
+    logAnalyticsEvent(
+      Events.vama_evidence_cont_3(
+        claim?.id || '',
+        request?.trackedItemId || null,
+        request?.type || 'Submit Evidence',
+        'file',
+      ),
+    )
     const mutateOptions = {
       onMutate: () => {
-        logAnalyticsEvent(Events.vama_claim_upload_start(claimID, request.trackedItemId || null, request.type, 'file'))
+        logAnalyticsEvent(
+          Events.vama_claim_upload_start(
+            claimID,
+            request?.trackedItemId || null,
+            request?.type || 'Submit Evidence',
+            'file',
+          ),
+        )
       },
       onSuccess: () => {
         setFilesList([])
         setFilesUploadedSuccess(true)
-        logAnalyticsEvent(Events.vama_claim_upload_compl(claimID, request.trackedItemId || null, request.type, 'file'))
+        logAnalyticsEvent(
+          Events.vama_claim_upload_compl(
+            claimID,
+            request?.trackedItemId || null,
+            request?.type || 'Submit Evidence',
+            'file',
+          ),
+        )
         showSnackBar(snackbarMessages.successMsg, dispatch, undefined, true)
       },
       onError: () => showSnackBar(snackbarMessages.errorMsg, dispatch, onUploadConfirmed, false, true),
     }
-    const params: UploadFileToClaimParamaters = { claimID, request, files: filesList }
+    const params: UploadFileToClaimParamaters = { claimID, documentType: documentType, request, files: filesList }
```

Comment:
Can use shorthand for documentType, same as other params

Reply:
updated